primitives/std.lib

```diff
@@ -870,11 +865,56 @@ primitive fixed_p_std_add_dbit[width, int_width1, fract_width1, int_width2, frac
           end
         end
 
-        //add
-        //assign whole_int = left_int + right_int;
-        //assign whole_fract = mod_left_fract + right_fract;
-
         assign out = {whole_int, whole_fract};
       endmodule
     }
+}
+
+/////// signed primitives ///////
+primitive std_uadd<"share"=1>[width](left: width, right: width) -> (out: width) {
+  verilog {
+    module std_uadd
+      #(parameter width = 32)
+      (input  signed  [width-1:0] left,
+        input  signed  [width-1:0] right,
+        output signed  [width-1:0] out);
+      assign out = left + right;
+    endmodule
+  }
+}
+
+primitive std_usub<"share"=1>[width](left: width, right: width) -> (out: width) {
+  verilog {
+    module std_usub
+      #(parameter width = 32)
+      (input  signed  [width-1:0] left,
+        input  signed  [width-1:0] right,
+        output signed  [width-1:0] out);
+      assign out = left - right;
+    endmodule
+  }
+}
+
+primitive std_umult<"share"=1>[width](left: width, right: width) -> (out: width) {
+  verilog {
+    module std_umult
+      #(parameter width = 32)
+      (input  signed  [width-1:0] left,
+        input  signed  [width-1:0] right,
+        output signed  [width-1:0] out);
+      assign out = left * right;
+    endmodule
+  }
+}
+
+primitive std_div<"share"=1>[width](left: width, right: width) -> (out: width) {
```

Comment:
I think you meant `std_udiv` here.

Reply:
good point I just fixed it

Reply:
Don't know why I saw this today, so sorry for the late response. (I'm not sure if you still need the answers but I'd respond anyways)I'll change them to signed since it is indeed confusing. I think only difference would be using signed instead of wires or logic in the Verilog code. For now I'm expecting that I don't need to explicitly add signed memory or registers since it's not the binary numbers that is effected but how they are interpreted within the system. I guess I would add new library contents for signed computations that are needed.